File: sdkcreationmw/sdkruntimes/MIDP/nei/src/com/symbian/tools/j2me/sei/emulator/session/commands/RemoveCommand.java
/*
* Copyright (c) 2003 - 2004 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description: 
*
*/

package com.symbian.tools.j2me.sei.emulator.session.commands;

import java.io.*;
import com.symbian.io.*;
import com.symbian.vm.*;


public class RemoveCommand extends SessionCommand {
  //
  // Constants
  //

  /**
   * Remove command identifier
   */
  private static final int REMOVE_CMD_ID = 2;

  //
  // Members
  //

  /**
   * MIDlet suite information
   */
  private final MIDletSuiteInfo iSuiteInfo;

  //
  // Life cycle
  //

  /**
   * Constructor
   *
   * @param aSuiteInfo suite info
   */
  public RemoveCommand(MIDletSuiteInfo aSuiteInfo) {
    super(REMOVE_CMD_ID);
    iSuiteInfo = aSuiteInfo;
  }

  //
  // Operations
  //

  /**
   * Dump the object to the output stream
   *
   * @param aDos Data output stream
   *
   * @throws IOException if some error occured
   */
  public void dump(DataOutputStream aDos) throws IOException {
    super.dump(aDos);
    aDos.write((byte)1); // "command exits" flag necessary to fulfill protocol
    aDos.writeUTF(iSuiteInfo.getSuiteName());
    aDos.writeUTF(iSuiteInfo.getVendor());
    aDos.flush();
  }

}

File: sdkcreationmw/sdkruntimes/MIDP/DebugAgent/debugagent_orig/src/seislave/commands/RemoveCommand.cpp
/*
* Copyright (c) 2004 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description: 
*
*/


#include "SeiCommands.h"

CRemoveCommand* CRemoveCommand::NewL(
    CDriver* aDriver, 
    CSocketReader* aReader,
    CSocketWriter* aWriter)
    {
    CRemoveCommand* object = new (ELeave) CRemoveCommand(aDriver, aReader, aWriter);
    CleanupStack::PushL(object);
    object->ConstructL();
    CleanupStack::Pop(object);
    return object;
    }

CRemoveCommand::CRemoveCommand(
    CDriver* aDriver, 
    CSocketReader* aReader,
    CSocketWriter* aWriter)
    : CSeiCommand(aDriver, aReader, aWriter)
    {
    }

CRemoveCommand::~CRemoveCommand()
    {
    if ( iName != NULL )
        {
        delete iName;
        }
    if ( iVendor != NULL )
        {
        delete iVendor;
        }
    }

void CRemoveCommand::ConstructL()
    {
    }

void CRemoveCommand::DoReadL()
    {
	iDriver->Log(_L("Reading remove MIDlet request"));
    TUint8 tmpFinal = iReader->ReadL();
    if (tmpFinal == 0)
        {
        iFinal = EFalse;
        }
    else 
        {
        iFinal = ETrue;
        }
    iReader->ReadUtfL(iName);
    iReader->ReadUtfL(iVendor);
    }

void CRemoveCommand::DoExecuteL()
    {
	iDriver->Log(_L("Removing MIDlet..."));
    iDriver->Engine()->RemoveL(iName, iVendor);
    }

void CRemoveCommand::DoWriteL()
    {
	iDriver->Log(_L("Remove MIDlet request complete"));
    iWriter->WriteIntL(KErrNone);
    }
    
TBool CRemoveCommand::isFinal()
	{
	return iFinal;
	}    
